### PR TITLE
Pipenv

### PR DESCRIPTION
**Pull request by kennethreitz**
Sending this PR on behalf of @digitalocean.

tl;dr: We should use modern tools, if available :)

This solves many problems, which should be evident, but I'll note some of them:

- seperating the "deps i want" vs. "deps i need" concerns
- declares the greater Python version in the `Pipfile` (Pipenv's default, but optional, behavior)
- uses Postgres instead of MySQL, which is far-and-away considered best practice within the Python community
- easier maintinence

`pip-tools` does solve many of these problems itself, and supporting `requirements.txt` should still be a target. Using https://buildpacks.io should make this relatively simple though, as Heroku supports both Pipenv and `requirements.txt` (and `setup.py`, secretly) at the same time. Pipenv is the documented and recommended tool, however.

### CODE REVIEW

**Comment by kennethreitz** [comment]
I also added a production web server (gunicorn) and pointed out how to run it with the `[scripts]` section. You could harness/parse this in your platform (`Pipfile` is a TOML file), or just use it as an example to update how it's executed elsewhere in your system.

**Comment by kennethreitz** [comment]
One question: do you want Windows users to be able to run this locally?

**Comment by kennethreitz** [comment]
If so, Gunicorn will need to be replaced with Waitress.

**Comment by kennethreitz** [comment]
Also, does your router prevent SlowLoRiS attacks (e.g. do HTTP buffering?). If not, waitress would be a better choice anyway.
